Makefile

```diff
@@ -78,11 +87,12 @@ test: test/assets
 	# declare -p HTTP_PROXY
 	$(PYTHON) -m pytest test $(PYTEST_ARGS)
 
+# Test the command line tools
 test-cli: test/assets
 	$(PIP) install -e .
 	rm -rfv test-workspace
 	cp -rv test/assets/kant_aufklaerung_1784 test-workspace
-	cd test-workspace/data && \
+	export LC_ALL=C.UTF-8; cd test-workspace/data && \
```

Comment:
C.f. https://circleci.com/gh/OCR-D/ocrd_tesserocr/27:

```
Traceback (most recent call last):
  File "/usr/local/bin/ocrd-tesserocr-segment-region", line 11, in <module>
    load_entry_point('ocrd-tesserocr', 'console_scripts', 'ocrd-tesserocr-segment-region')()
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 696, in main
    _verify_python3_env()
  File "/usr/local/lib/python3.6/dist-packages/click/_unicodefun.py", line 124, in _verify_python3_env
    ' mitigation steps.' + extra
RuntimeError: Click will abort further execution because Python 3 was configured to use ASCII as encoding for the environment. Consult https://click.palletsprojects.com/en/7.x/python3/ for mitigation steps.

This system supports the C.UTF-8 locale which is recommended.
You might be able to resolve your issue by exporting the
following environment variables:

    export LC_ALL=C.UTF-8
    export LANG=C.UTF-8
```

I will debug this further. Since I cannot ssh into the Circle CI containers (yay firewall), expect some force pushes.

Reply:
oops. I can take that if you like. I remember this error was the very reason that the locale bridge had to be inserted __after__ click but __before__ any tesserocr stuff. Obviously this is not the case anymore!

Reply:
Problem is that `LC_ALL` is unset in the CI environment. When locale is unset, python3 apparently falls back to ASCII which upsets click.

We could hack the console scripts with the same trick you used in https://github.com/OCR-D/ocrd_tesserocr/commit/1ba854016e0651e93b94b7c060314a533bf0446d but that would be a lot of effort for little gain.

I suggest to just document that users have to set their locale before running any of the ocrd tools...

@wrznr

Reply:
Indeed, this is not at all about Tesseract's ([now removed](https://github.com/tesseract-ocr/tesseract/pull/2437/files) dependency on some `C` or `C.UTF-8` locale), but about [Click itself in Python3](https://click.palletsprojects.com/en/7.x/python3/#python3-surrogates) with non-UTF8 (including empty) settings for `LC_CTYPE` or `LC_ALL`.

> We could hack the console scripts with the same trick you used in [1ba8540](https://github.com/OCR-D/ocrd_tesserocr/commit/1ba854016e0651e93b94b7c060314a533bf0446d) but that would be a lot of effort for little gain.

I can't see that. Doing that trick with Click (not with Tesseract) does not seem to work. 

But I don't think it's a good idea to require the _user_  to always ensure UTF-8 l10n in the shell, either. (While we can, of course, easily amend the dockerfiles we control.) This is bound to fail unexpectedly. 

Interestingly, the problem goes away with Python 3.7 onwards (as the above documentation confirms). So maybe we can import something from `future` in core?

Reply:
Either way, I think that resolves this. Maybe we can do the shell exports in `ocrd/core` and `ocrd/tesserocr` dockerfiles instead in a separate PR?

Reply:
It's not trivial to implement with `future` or hacks at the top of the program, because that behavior is part of the python interpreter startup. Python 3.7 basically sets `LC_ALL` to `C.UTF-8`, if the locale is `C` or unset, just what we want. There are efforts to backport this to python 3.6 (e.g. https://fedoraproject.org/wiki/Changes/python3_c.utf-8_locale#Current_status) but you cannot rely on that. I find it easier to explain to users to set a UTF-8 locale than to point them to the right patch version release of their python interpreter or explain how building and patching python works.

BTW I wonder if there is any reason to set a terminal's encoding to anything other than UTF-8 in 2019. My gut feeling is that very few users deliberately do not want a UTF-8 encoding.

Reply:
Oh I see. Agreed, documentation as a required setting is better. (And thankfully, Click's self-documentation is quite good already.)

> BTW I wonder if there is any reason to set a terminal's encoding to anything other than UTF-8 in 2019. My gut feeling is that very few users deliberately do not want a UTF-8 encoding.

There's lots of old code in use out there. I still use 8-bit l10ns myself from time to time (I even wrote tty+terminal switch commands for it). And then there are things like GB18030, which is the PR China's UTF standard, and widely used...